src/pages/subjects/Subjects.tsx

```diff
@@ -78,9 +78,9 @@ const Subjects = () => {
   }
 
   const getSubjects = useCallback(
-    (data?: Pick<SubjectInterface, 'name'>) =>
-      subjectService.getSubjects(data, categoryId),
-    [categoryId]
+    (data?: Pick<SubjectInterface, 'name' | 'category'>) =>
```

Comment:
Why is it optional here? You replaced useAxios with useQuery here, so optionality is redundant here
```suggestion
    (data: Pick<SubjectInterface, 'name' | 'category'>) =>
```

Reply:
Yeah, true. Fixed.